src/apps/vendor-finance/common/vendor-finance.pool.contract-position-fetcher.ts

```diff
@@ -149,12 +143,18 @@ export abstract class VendorFinancePoolContractPositionFetcher extends ContractP
     // --! Lender logic !---
 
     // --- Borrower logic ----
-    const [borrowPosition, mintRatioRaw] = await Promise.all([contract.debt(address), contract.mintRatio()]);
+    const data: VendorBorrowerGraphResponse = await Axios.post(this.entityUrl, {
+      query: borrowerInfosQuery(address),
+      type: `v1-${this.network.charAt(0).toUpperCase() + this.network.slice(1)}`,
+    });
+
+    const borrowerPosition = data.data.borrower?.positions.find(({ pool }) => pool.id === contractPosition.address);
+    if (!borrowerPosition) return [];
```

Comment:
Doesn't `contract.debt(address) ` return the borrow position ?

Reply:
Sorry, got bogged down with something earlier. 

Yes you are right. I'm not entirely sure what prompted me to make the unfavorable switch in the first place. My mistake. I know you want to move away from external API calls whenever possible so I'll be more cognizant of that.